boot/build.go

```diff
@@ -185,87 +155,120 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 		classpathLayer.Logger = b.Logger
 		result.Layers = append(result.Layers, classpathLayer)
 
-	} else {
-		scbJarFound := FindExistingDependency(d, "spring-cloud-bindings")
-		if scbJarFound {
-			b.Logger.Header("A Spring Cloud Bindings library was found in the Spring Boot libs - not adding another one")
-		}
-		// contribute Spring Cloud Bindings - false by default
-		if !cr.ResolveBool("BP_SPRING_CLOUD_BINDINGS_DISABLED") && !scbJarFound {
-			h, be := libpak.NewHelperLayer(context.Buildpack, "spring-cloud-bindings")
-			h.Logger = b.Logger
-			result.Layers = append(result.Layers, h)
-			result.BOM.Entries = append(result.BOM.Entries, be)
-
-			scbVer, scbSet := cr.Resolve("BP_SPRING_CLOUD_BINDINGS_VERSION")
-			if !scbSet {
-				scbVerFromBoot, err := getSCBVersion(version)
-				if err != nil {
-					return libcnb.BuildResult{}, fmt.Errorf(
-						"unable to read the Spring Boot version from META-INF/MANIFEST.MF. " +
-							"Please set BP_SPRING_CLOUD_BINDINGS_VERSION to force a version or " +
-							"BP_SPRING_CLOUD_BINDINGS_DISABLED to bypass installing Spring Cloud Bindings")
-				}
-				scbVer = scbVerFromBoot
-			}
+		return result, nil
 
-			dep, err := dr.Resolve("spring-cloud-bindings", scbVer)
-			if err != nil {
-				return libcnb.BuildResult{}, fmt.Errorf("unable to find dependency\n%w", err)
-			}
+	}
 
-			bindingsLayer, be := NewSpringCloudBindings(filepath.Join(context.Application.Path, lib), dep, dc)
-			bindingsLayer.Logger = b.Logger
-			result.Layers = append(result.Layers, bindingsLayer)
-			result.BOM.Entries = append(result.BOM.Entries, be)
-		}
+	// Spring Cloud Bindings
+	if scbJarFound := FindExistingDependency(d, "spring-cloud-bindings"); scbJarFound {
+		b.Logger.Header("A Spring Cloud Bindings library was found in the Spring Boot libs - not adding another one")
+	} else if !cr.ResolveBool("BP_SPRING_CLOUD_BINDINGS_DISABLED") {
 
-		// configure JVM for application type
-		classes, ok := manifest.Get("Spring-Boot-Classes")
-		if !ok {
-			return libcnb.BuildResult{}, fmt.Errorf("manifest does not contain Spring-Boot-Classes")
+		var scbVer string
+		var scbSet bool
+		if scbVer, scbSet = cr.Resolve("BP_SPRING_CLOUD_BINDINGS_VERSION"); !scbSet {
+			if scbVer, err = getSCBVersion(version); err != nil {
+				return libcnb.BuildResult{}, fmt.Errorf(
+					"unable to read the Spring Boot version from META-INF/MANIFEST.MF. " +
```

Comment:
this part of the error message is actually wrong. It should read:
"unable to parse the Spring Boot version in the META-INF/MANIFEST.MF. "

it's maybe a detail, but by this point, we were able to read the SB version for sure, we just did not recognize it